code/modules/mob/living/simple_animal/guardian/types/charger.dm

```diff
@@ -51,24 +51,20 @@
 /mob/living/simple_animal/hostile/guardian/charger/throw_impact(atom/hit_atom, datum/thrownthing/throwingdatum)
 	if(!charging)
 		return ..()
-
-	else if(hit_atom)
-		if(isliving(hit_atom) && hit_atom != summoner)
-			var/mob/living/L = hit_atom
-			var/blocked = FALSE
-			if(hasmatchingsummoner(hit_atom)) //if the summoner matches don't hurt them
-				blocked = TRUE
-			if(ishuman(hit_atom))
-				var/mob/living/carbon/human/H = hit_atom
-				if(H.check_shields(src, 90, "[name]", attack_type = THROWN_PROJECTILE_ATTACK))
-					blocked = TRUE
-			if(!blocked)
-				L.drop_all_held_items()
-				L.visible_message(span_danger("[src] slams into [L]!"), span_userdanger("[src] slams into you!"))
-				L.apply_damage(20, BRUTE)
-				playsound(get_turf(L), 'sound/effects/meteorimpact.ogg', 100, TRUE)
-				shake_camera(L, 4, 3)
-				shake_camera(src, 2, 3)
-
-		charging = 0
+	if(!isliving(hit_atom) || hit_atom == summoner)
+		return
```

Comment:
isn't there a slight behavior change here? if we hit a non valid thing, we sued to set charging to false, but now we don't.
actually charging's just gone is it event based or something?

Reply:
in charging_end we already set charging to false